lib/origen/specs.rb

```diff
@@ -324,6 +332,53 @@ def doc_resources(options = {})
       end
     end
 
+    def documentations(options = {})
+      options = {
+        section:    nil,
+        subsection: nil,
+        interface:  nil,
+        mode:       nil,
+        type:       nil,
+        sub_type:   nil,
+        audience:   nil
+      }.update(options)
+      return nil if @_documentation.nil?
+      return nil if @_documentation.empty?
+      doc_found = Hash.new do |h, k|
+        h[k] = Hash.new do |hh, kk|
+          hh[kk] = Hash.new do |hhh, kkk|
+            hhh[kkk] = Hash.new do |hhhh, kkkk|
+              hhhh[kkkk] = Hash.new do |hhhhh, kkkkk|
+                hhhhh[kkkkk] = Hash.new do |hhhhhh, kkkkkk|
+                  hhhhhh[kkkkkk] = {}
```

Comment:
Yikes!
Don't need to change it right now, but this trick should help with this kind of thing:
https://github.com/franzejr/best-ruby/blob/master/tricks/autovivification.md